cpp/src/groupby/hash/groupby.cu

```diff
@@ -279,27 +279,19 @@ void compute_single_pass_aggs(table_view const& keys,
 
   bool skip_key_rows_with_nulls = keys_have_nulls and include_null_keys == null_policy::EXCLUDE;
 
-  if (skip_key_rows_with_nulls) {
-    auto row_bitmask{cudf::detail::bitmask_and(keys, stream)};
-    thrust::for_each_n(
-      rmm::exec_policy(stream)->on(stream.value()),
-      thrust::make_counting_iterator(0),
-      keys.num_rows(),
-      hash::compute_single_pass_aggs<true, Map>{map,
-                                                keys.num_rows(),
-                                                *d_values,
-                                                *d_sparse_table,
-                                                d_aggs.data().get(),
-                                                static_cast<bitmask_type*>(row_bitmask.data())});
-  } else {
-    thrust::for_each_n(
-      rmm::exec_policy(stream)->on(stream.value()),
-      thrust::make_counting_iterator(0),
-      keys.num_rows(),
-      hash::compute_single_pass_aggs<false, Map>{
-        map, keys.num_rows(), *d_values, *d_sparse_table, d_aggs.data().get(), nullptr});
-  }
-
+  auto row_bitmask =
```

Comment:
If `skip_key_rows_with_nulls` is only used in this function, why not make it a parameter of this function rather than a member variable of the struct?

Reply:
This is a functor whose `operator()` is called by `thrust::for_each`
Everything the `operator()` needs is a member variable of the functor.

Reply:
OK, thought it was type-dispatch functor.